jest/jest-setup-after-env.ts

```diff
@@ -2,77 +2,93 @@ import { afterAll, afterEach, beforeAll, beforeEach, expect, jest } from '@jest/
 // Polyfills `window.fetch` for Jest because it runs in a Node environment where fetch isn’t available. It initially looked like this would change with Node.js 18, but that is not so.
 import 'isomorphic-fetch'
 import { config } from '@vue/test-utils'
-import { rest, MockedRequest as Request } from 'msw'
 import { setupServer } from 'msw/node'
 
 import { replaceAttributesSnapshotSerializer } from './jest-replace-attribute-snapshot-serializer'
 import { TOKENS as COMPONENT_TOKENS } from '../src/components'
-import { createRouter } from '../src/router/router'
-import { TOKENS, get, container, build } from '../src/services/development'
+import { TOKENS, get, container, build, merge, createInjections } from '../src/services/development'
+import { token } from '../src/services/utils'
 import Env from '@/services/env/Env'
-
-type MockFunction = (_opts: Record<string, unknown>, cb: (req: Request, resp: Record <string, any>) => Record<string, unknown>) => void
+import { mocker } from '@/test-support'
+import type { Mocker } from '@/test-support'
+import type { RestHandler } from 'msw'
 
 // jest can't import this module properly due to transpiling issues
 // mock this out with a blank element
 jest.mock('vue-github-button', () => ({ template: '<span />' }))
+//
 
-/**
- * Adds the application’s router to vue test utils. This way tests don’t have to set-up a new router instance on their own.
- */
-const router = createRouter(get(TOKENS.routes), get(TOKENS.store))
-config.global.plugins.push(router)
+const $ = {
+  ...TOKENS,
+  mock: token<Mocker>('mocker'),
+};
 
-/**
- * Kongponents uses generated UUIDs for several attribute values.
- * This breaks the project’s snapshot tests since they’re based on fully-mounted components
- * which also includes those from external sources like Kongponents.
- *
- * In order to stabilize the tests which otherwise fail because the attribute values are different every run,
- * we use a custom snapshot serializer to replace those attribute values with one fixed value.
- */
-expect.addSnapshotSerializer(replaceAttributesSnapshotSerializer([
-  'id',
-  'aria-describedby',
-  'aria-labelledby',
-  'aria-controls',
-  'data-tableid',
-]))
+(async () => {
+  const services = merge([
+    [$.msw, {
```

Comment:
**Suggestion**: I’d prefer this service to be called `mswServer` and the worker in development.ts `mswWorker` to not create the illusion that this service is interchangeable. While they serve the same purpose, they’re functionally different and won’t ever be able to be used in the respective other environment (i.e. you can’t use the MSW server in a browser context and you can’t use the MSW worker in a Node.js context) so there isn’t a benefit to making them the same service.

Reply:
What happens if you have application code that does `get($.msw)` in a browser _or_ CLI. Do you have to add application code to detect whether you are in a browser or node and call `get($.mswWorker)` or `get($.mswServer)` accordingly?

Alternatively (as we do here) we keep it as `get($.msw)` and inject a different implementation from The Outside?

Reply:
> Where would you need that in the first place? MSW is not used in application code but as part of the setup of the individual testing or development environments.

We use it here, which I would count as application code:

https://github.com/kumahq/kuma-gui/blob/1972104661ae1dd4ace8a0b46fc4af9460a1e099/src/services/development.ts#L97-L107

If I were to run this code in a browser I want it to get the worker, whereas if I run it in some node CLI based tests I want it to return the equivalent in node i.e. the node version of the worker.

Should I add a conditional to detect whether this is browser or node here? Or should I also overwrite `bootstrap` yet again during testing? Alternatively (as we do here) we keep it as `get($.msw)` and inject a different implementation from The Outside?

Reply:
Ok thanks for the viewpoint, if it becomes problematic I'll happily change it.

Lemme know if theres anything else in the overall PR.